### ISSUE TITLE
CLI: expose training job status and a minimal job list (jammi train status / jammi train list)

### ISSUE THREAD

**Issue by vijaydevmasters** [f-inverse/jammi-ai#344]
## Summary

There is no `jammi` CLI command to list running or completed training jobs, or to poll the current epoch and loss for an in-progress job. The `training_jobs` catalog table exists server-side but is completely unexposed through the CLI.

## Observed behaviour

```
$ jammi --help
Commands:
  status    Report the server's capabilities
  sources   Manage data sources
  models    Manage models
  trigger   Manage trigger-stream topics
  channels  Manage evidence channels
  mutable   Manage mutable companion tables
```

There is no `train` subcommand. Users who submit a `fine_tune` job via the Python SDK or gRPC have no way to check progress without writing custom queries against the catalog directly.

## Expected behaviour

```
$ jammi train list
JOB_ID           STATUS     EPOCH  STEP   TRAIN_LOSS  VAL_LOSS  CREATED
---------------  ---------  -----  -----  ----------  --------  -------------------
3f2e1d...        running    4/10   12400  0.312       0.341     2026-08-13 14:02
a1b2c3...        completed  10/10  19810  0.198       0.221     2026-08-12 09:15

$ jammi train status 3f2e1d...
job_id:     3f2e1d...
status:     running
epoch:      4 / 10
step:       12400 / 19810
train_loss: 0.312
val_loss:   0.341
model_id:   bioclinical-modernbert-large-lora-v1
started_at: 2026-08-13 14:02:16 UTC
```

## Suggested implementation

- Add a `train` top-level subcommand to `crates/jammi-cli/src/main.rs` with `list` and `status <JOB_ID>` actions, mirroring the `sources` / `models` pattern.
- Expose `list_training_jobs` and `describe_training_job` verbs on `CatalogClient` in `crates/jammi-admin`, backed by the existing `training_repo`.
- The `status` view should include current epoch, global step, latest train/val loss, and elapsed time — data the trainer already writes to the catalog on each epoch boundary.

**Comment by thankrandomness** [comment]
Rescoped (2026-08-30) after verification against main @ c0a9e63 — the original ask bundled three different things:

**In scope (this issue):**
1. `jammi train status <JOB_ID>` — a CLI verb over the existing `TrainingStatus` RPC (status / model_id / error). The wire surface and server already exist; the missing pieces are the `jammi-admin` client method and the CLI subcommand.
2. `jammi train list` — a minimal `ListTrainingJobs` RPC (job_id, status, base/output model ids, timestamps) backed by the `training_jobs` catalog table, tenant-scoped like every other RPC, plus the CLI verb. No `ListTrainingJobs` RPC exists today.

**Out of scope (platform monitoring, not the OSS engine):** the originally sketched live progress columns (epoch N/M, step, train_loss, val_loss). This is not an oversight but an architectural position: the worker persists run metrics only at the finalize CAS — mid-run, progress exists solely in the worker's memory — and `training.proto` states "There is no progress stream" by design. Live progress therefore requires a persistence-model change or a progress stream, which is production-monitoring territory (jammi-enterprise), not engine mechanism.

For reference, the Python SDK already exposes lifecycle polling (`job.status()", `job.wait()"); this issue brings the CLI to parity with that, no further.

**Comment by thankrandomness** [comment]
Delivered in #419 (merged): `jammi train status <JOB_ID>` and `jammi train list` — the additive `ListTrainingJobs` RPC (freeze-guard baseline updated per its additive flow), tenant-scoped server handler with the cross-tenant listing case registered in the tenant-isolation oracle, candle-free read verbs on `jammi-admin`, and the CLI subcommand, all verified end-to-end against a live server with tenant isolation pinned in both directions on the wire. Scope per the rescope above: lifecycle-only by design — live epoch/step/loss progress requires a persistence-model change and is platform monitoring, not engine surface.
